payments_service/braintree/views.py

```diff
@@ -129,22 +135,114 @@ def render(self, data, media_type=None, renderer_context=None):
         return data.encode(self.charset)
 
 
-class Webhook(UnprotectedAPIView, SolitudeBodyguard):
-    methods = ['post', 'get']
-    resource = 'braintree.webhook'
+class Webhook(UnprotectedAPIView):
+    renderer_classes = [PlainTextRenderer]
 
-    def perform_content_negotiation(self, request, **kw):
-        """
-        Braintree sends a request that has an Accept Header of
-        u'*/*; q=0.5', u'application/xml', but we need to force
-        DRF to return it as text/plain. The only way we can do that
-        is to override the content negotiation and insert our rather
-        boring plain text renderer.
+    def __init__(self, *args, **kw):
+        super(Webhook, self).__init__(*args, **kw)
+        self.api = solitude.api()
 
-        See https://github.com/braintree/braintree_python/issues/54
-        """
-        if request.method.lower() == 'get':
-            renderer = PlainTextRenderer()
-            return (renderer, renderer.media_type)
-        return (super(self.__class__, self)
-                .perform_content_negotiation(request, **kw))
+    def get(self, request):
+        try:
+            result = self.api.braintree.webhook.get(
+                bt_challenge=request.REQUEST.get('bt_challenge'))
+            status = 200
+        except HttpClientError, exc:
+            log.info('webhook GET: ignoring bad request: '
+                     '{e.__class__.__name__}: {e}'.format(e=exc))
+            result = 'verification failed'
+            status = 400
+        return Response(result, status=status,
+                        content_type='text/plain; charset=utf-8')
+
+    def post(self, request):
+        try:
+            webhook_result = self.api.braintree.webhook.post({
+                'bt_payload': request.DATA.get('bt_payload'),
+                'bt_signature': request.DATA.get('bt_signature'),
+            })
+            self.notify_user_of_webhook(webhook_result)
+            result = ''
+            status = 200
+        except HttpClientError, exc:
+            log.info('webhook POST: ignoring bad request: '
+                     '{e.__class__.__name__}: {e}'.format(e=exc))
+            result = 'verification failed'
+            status = 400
+
+        # Even though we might be returning an empty response,
+        # Braintree expects this to be a 200.
+        return Response('', status=status)
+
+    def notify_user_of_webhook(self, result):
+        log.debug('about to handle subscription: {s}'
+                  .format(s=result))
+
+        kind = result['braintree']['kind']
+        if kind !=  'subscription_charged_successfully':
+            log.warning('not notifying user of webhook kind: {k}'
+                        .format(k=kind))
+            return
```

Comment:
I was making the assumption that if the response is a 204, you can ignore it. Might help a little: http://solitude.readthedocs.org/en/latest/topics/braintree.html#post--braintree-webhook-